src/include/ndpi_typedefs.h

```diff
@@ -1551,6 +1551,8 @@ struct ndpi_flow_struct {
 
     struct {
       char filename[128];
+      u_int16_t data_num;
+      u_int16_t ack_num;
```

Comment:
This is wrong. These two variables need to be defined in `struct ndpi_flow_udp_struct` (like `tftp_stage`).
Reason: since `ndpi_flow_struct->protos` is an union, we can't write to it until we have a final classification, i.e. until we call `ndpi_int_tftp_add_connection()`.

Could you rebase and update the code, please?